great_expectations/data_context/_version_checker.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+import json
+import logging
+
+import requests
+from packaging import version
+from typing_extensions import TypedDict
+
+logger = logging.getLogger()
+
+
+class _PyPIPackageInfo(TypedDict):
+    version: str
+
+
+class _PyPIPackageData(TypedDict):
+    info: _PyPIPackageInfo
+
+
+class _VersionChecker:
+
+    _BASE_PYPI_URL = "https://pypi.org/pypi"
+    _PYPI_GX_ENDPOINT = f"{_BASE_PYPI_URL}/great_expectations/json"
+
+    def __init__(self, user_version: str) -> None:
+        self._user_version = version.Version(user_version)
+
+    def check_if_using_latest_gx(self) -> None:
+        pypi_version = self._get_latest_version_from_pypi()
+        if not pypi_version:
+            logger.debug("Could not compare with latest PyPI version; skipping check.")
+            return
+
+        if self._is_using_outdated_release(pypi_version):
+            self._warn_user(pypi_version)
+
+    def _get_latest_version_from_pypi(self) -> version.Version | None:
+        response_json: _PyPIPackageData | None = None
+        try:
+            response = requests.get(self._PYPI_GX_ENDPOINT)
```

Comment:
It might be worth it to create a `responses` fixture mock for this.
Or... adding a check here to see if we the code is being run from pytest.

https://stackoverflow.com/a/44595269/6304433

Reply:
ooh might ask for some help here to make sure no other tests are impacted - will probably tackle tomorrow